File: trunk/java/SR-RBD/src/ddb/tpc/msg/CanCommitMessage.java
/**
 * 
 */
package ddb.tpc.msg;

import java.io.IOException;

import ddb.communication.DataInputStream;
import ddb.communication.DataOutputStream;
import ddb.msg.MessageType;

/** 
 * <!-- begin-UML-doc -->
 * <!-- end-UML-doc -->
 * @author User
 * @generated "UML to Java (com.ibm.xtools.transform.uml2.java5.internal.UML2JavaTransform)"
 */
public class CanCommitMessage extends TPCMessage {
	/** 
	 * <!-- begin-UML-doc -->
	 * Zapytanie klienta przerobione przez parser sql.
	 * <!-- end-UML-doc -->
	 * @generated "UML to Java (com.ibm.xtools.transform.uml2.java5.internal.UML2JavaTransform)"
	 */
	private String queryString;
	/** 
	 * <!-- begin-UML-doc -->
	 * Nazwa tabeli jakiej dotyczy zapytanie.
	 * <!-- end-UML-doc -->
	 * @generated "UML to Java (com.ibm.xtools.transform.uml2.java5.internal.UML2JavaTransform)"
	 */
	private String tableName;
	/**
	 * Czy transakcja dotyczy utworzenia nowej tabeli
	 */
	private boolean isCreate;
	
	/** 
	 * <!-- begin-UML-doc -->
	 * <!-- end-UML-doc -->
	 * @return
	 * @generated "UML to Java (com.ibm.xtools.transform.uml2.java5.internal.UML2JavaTransform)"
	 */
	public String getTableName() {
		return this.tableName;
	}

	/** 
	 * <!-- begin-UML-doc -->
	 * <!-- end-UML-doc -->
	 * @param tableName
	 * @generated "UML to Java (com.ibm.xtools.transform.uml2.java5.internal.UML2JavaTransform)"
	 */
	public void setTableName(String tableName) {
		this.tableName = tableName;
	}

	/** 
	 * <!-- begin-UML-doc -->
	 * <!-- end-UML-doc -->
	 * @param queryString
	 * @generated "UML to Java (com.ibm.xtools.transform.uml2.java5.internal.UML2JavaTransform)"
	 */
	public void setQueryString(String queryString) {
		this.queryString = queryString;
	}

	/** 
	 * <!-- begin-UML-doc -->
	 * <!-- end-UML-doc -->
	 * @return
	 * @generated "UML to Java (com.ibm.xtools.transform.uml2.java5.internal.UML2JavaTransform)"
	 */
	public String getQueryString() {
		return  queryString;
	}
	
	public boolean isCreate() {
		return isCreate;
	}

	public void setCreate(boolean isCreate) {
		this.isCreate = isCreate;
	}

	@Override
	public void fromBinary(DataInputStream s) throws IOException {
		queryString = s.readString();
		tableName = s.readString();
		isCreate = s.readBoolean();
	}

	@Override
	public MessageType getType() {
		return MessageType.TPC_CANCOMMIT;
	}

	@Override
	public void toBinary(DataOutputStream s) throws IOException {
		s.writeString(queryString);
		s.writeString(tableName);
		s.writeBoolean(isCreate);
	}
}

File: trunk/C#/RBD/RBD/TPC/Msg/CanCommitMessage.cs
﻿// + TODO check

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using RBD.Msg;
using RBD.Communication;

namespace RBD.TPC.Msg
{
    public class CanCommitMessage : TPCMessage
    {
        public String QueryString { get; set; }
        public String TableName { get; set; }
        /**
         * Czy transakcja dotyczy utworzenia nowej tabeli
         */
        public bool IsCreate { get; set; }
        /**
         * Numer wersji tabeli na koordynatorze
         */
        public int TableVersion { get; set; }

        override
        public void FromBinary(DataInputStream s) //throws IOException 
        {
            base.FromBinary(s);
            QueryString = s.ReadString();
            TableName = s.ReadString();
            //http://java.sun.com/j2se/1.4.2/docs/api/java/io/DataInput.html#readBoolean()
            IsCreate = s.ReadBoolean(); // 1 Byte
            TableVersion = s.ReadInt32();
        }

        override
        public MessageType GetMessageType()
        {
            return MessageType.TPC_CANCOMMIT;
        }

        override
        public void ToBinary(DataOutputStream s) //throws IOException 
        {
            base.ToBinary(s);
            s.WriteString(QueryString);
            s.WriteString(TableName);
            s.Write((bool)IsCreate);
            s.Write((int)TableVersion);
        }
    }
}
